pkg/controller/scylladbdatacenter/resource.go

```diff
@@ -213,17 +239,34 @@ func servicePorts(cluster *scyllav1.ScyllaCluster) []corev1.ServicePort {
 		},
 	}
 
-	if cluster.Spec.Alternator != nil {
+	if sdc.Spec.ScyllaDB.AlternatorOptions != nil {
 		ports = append(ports, corev1.ServicePort{
 			Name: alternatorTLSPortName,
 			Port: alternatorTLSPort,
 		})
 
-		enableHTTP := cluster.Spec.Alternator.InsecureEnableHTTP
-		if cluster.Spec.Alternator.Port != 0 || (enableHTTP != nil && *enableHTTP) {
+		var alternatorPort int
+		var enableHTTP bool
+		var err error
+
+		alternatorPortAnnotation, ok := sdc.Annotations[naming.TransformScyllaClusterToScyllaDBDatacenterAlternatorPortAnnotation]
+		if ok {
+			alternatorPort, err = strconv.Atoi(alternatorPortAnnotation)
+			if err != nil {
+				return nil, fmt.Errorf("can't parse alternator port annotation %q: %w", alternatorPortAnnotation, err)
+			}
+		}
+		enableHTTPAnnotation, ok := sdc.Annotations[naming.TransformScyllaClusterToScyllaDBDatacenterInsecureEnableHTTPAnnotation]
+		if ok {
+			enableHTTP, err = strconv.ParseBool(enableHTTPAnnotation)
+			if err != nil {
+				return nil, fmt.Errorf("can't parse enable http annotation %q: %w", enableHTTPAnnotation, err)
+			}
+		}
```

Comment:
nit: would be nice to make these into helper functions so they can be reused and test them

Reply:
there are unit tests targeted at these branches. I don't see any benefits of extracting it, these won't be used elsewhere